### PR TITLE
PixbufExt: allow creating a surface without a for_window

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Per the C API docs here https://developer.gnome.org/gdk3/stable/gdk3-Cairo-Interaction.html
gdk_cairo_surface_create_from_pixbuf the for_window param can be NULL

### CODE REVIEW

**Comment by sdroege** [comment]
:+1:

**Comment by bilelmoussaoui** [comment]
I'm kinda lost on how can I fix this issue
```
  --> src/cairo_interaction.rs:37:17
   |
37 |                 for_window.to_glib_none().0,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `gdk_sys::GdkWindow`, found associated type
   |
   = note: expected type `*mut gdk_sys::GdkWindow`
              found type `*mut <W as glib::ObjectType>::GlibType`
   = note: consider constraining the associated type `<W as glib::ObjectType>::GlibType` to `gdk_sys::GdkWindow`
   = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
```

File: src/cairo_interaction.rs [hunk 1]
```diff
@@ -25,16 +25,16 @@ impl GdkSurfaceExt for Surface {
...
 
 impl GdkPixbufExt for Pixbuf {
-    fn create_surface<W: IsA<Window>>(&self, scale: i32, for_window: &W) -> Option<Surface> {
+    fn create_surface<W: IsA<Window>>(&self, scale: i32, for_window: Option<&W>) -> Option<Surface> {
         unsafe {
             from_glib_full(gdk_sys::gdk_cairo_surface_create_from_pixbuf(
                 self.to_glib_none().0,
                 scale,
-                for_window.as_ref().to_glib_none().0,
+                for_window.to_glib_none().0,
```
Comment (EPashkin) [inline src/cairo_interaction.rs]: Seems is neededed something like `.map(std::convert::AsRef::as_ref)` co convert type
```
error[E0308]: mismatched types
  --> src\cairo_interaction.rs:37:17
   |
37 |                 for_window.to_glib_none().0,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `gdk_sys::GdkWindow`, found associated type
   |
   = note: expected type `*mut gdk_sys::GdkWindow`
              found type `*mut <W as glib::ObjectType>::GlibType`
   = note: consider constraining the associated type `<W as glib::ObjectType>::GlibType` to `gdk_sys::GdkWindow`
   = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
error: aborting due to previous error
```

And some `cargo fmt` too

Comment (bilelmoussaoui) [reply src/cairo_interaction.rs]: Thanks, should be fixed now

**Comment by EPashkin** [comment]
@bilelmoussaoui Thanks.
It not necessary, but can you change versions.txt too: we forgot to update it when upgraded submodule: it must contain "/gir @ d1e88f9", or I just restart osx builds.

**Comment by GuillaumeGomez** [comment]
Looks good to me as well, thanks! Please just do as @EPashkin suggested to fix the failure and we're good to go.

**Comment by bilelmoussaoui** [comment]
Done, thanks for the review :+1:

**Comment by GuillaumeGomez** [comment]
I'll very likely forget about this PR so once the CI is green, please ping me. :)

**Comment by EPashkin** [comment]
Seems OSx image changed in bad way.
@GuillaumeGomez I propose to merge this PR and solve this problem separately.

**Comment by GuillaumeGomez** [comment]
If OSX has an issue then yes, better to fix it in another PR.

**Comment by EPashkin** [comment]
CI passed except of OSX

**Comment by GuillaumeGomez** [comment]
Thanks for the update @EPashkin !

**Comment by bilelmoussaoui** [comment]
Thanks for the merge @GuillaumeGomez!

File: src/cairo_interaction.rs [hunk 2]
```diff
@@ -25,16 +25,21 @@ impl GdkSurfaceExt for Surface {
...
+        &self,
+        scale: i32,
+        for_window: Option<&W>,
+    ) -> Option<Surface> {
         unsafe {
             from_glib_full(gdk_sys::gdk_cairo_surface_create_from_pixbuf(
                 self.to_glib_none().0,
                 scale,
-                for_window.as_ref().to_glib_none().0,
+                for_window.map(std::convert::AsRef::as_ref).to_glib_none().0,
```
Comment (sdroege) [inline src/cairo_interaction.rs]: Why not `for_window.map(AsRef::as_ref).to_glib_none().0`? `AsRef` is in the prelude

Comment (bilelmoussaoui) [reply src/cairo_interaction.rs]: My bad, I didn't notice that AsRef was already in the prelude

**Pull request merged**